plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -192,79 +199,54 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
     });
   }
 
-  private fetchServicesByServiceId(
-    serviceId: string,
+  private fetchServicesForService(
     clusterDetails: ClusterDetails,
+    labelSelector: string,
   ): Promise<Array<V1Service>> {
     return this.singleClusterFetch<V1Service>(clusterDetails, ({ core }) =>
-      core.listServiceForAllNamespaces(
-        false,
-        '',
-        '',
-        `backstage.io/kubernetes-id=${serviceId}`,
-      ),
+      core.listServiceForAllNamespaces(false, '', '', labelSelector),
     );
   }
 
-  private fetchPodsByServiceId(
-    serviceId: string,
+  private fetchPodsForService(
     clusterDetails: ClusterDetails,
+    labelSelector: string,
   ): Promise<Array<V1Pod>> {
     return this.singleClusterFetch<V1Pod>(clusterDetails, ({ core }) =>
-      core.listPodForAllNamespaces(
-        false,
-        '',
-        '',
-        `backstage.io/kubernetes-id=${serviceId}`,
-      ),
+      core.listPodForAllNamespaces(false, '', '', labelSelector),
```

Comment:
yeah for real, this is the kind of place where it's super helpful for a reader if this were instead an object - but this is not our code right